client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/QueryOperation.java

```diff
@@ -86,7 +86,7 @@ protected QueryResponse executeOperation(Transport transport) {
             EnumMarshaller encoder = (EnumMarshaller) remoteQuery.getSerializationContext().getMarshaller(value.getClass());
             value = encoder.encode((Enum) value);
          } else if (value instanceof Boolean) {
-            value = (Boolean) value ? 1 : 0;
+            value = value.toString();
```

Comment:
From Lucene perspective, having indexing a boolean as two different terms instead of a number is a better choice. Indexing it as a number implies storing the field as a trie structure, and it requires more work to query it.